aten/src/ATen/native/TensorShape.cpp

```diff
@@ -840,6 +840,23 @@ Tensor repeat(const Tensor& self, IntArrayRef repeats) {
   return result;
 }
 
+Tensor tile(const Tensor& self, IntArrayRef reps){
+  // If self.size() > len(reps), reps is promoted to self.size() by pre-pending
+  // 1’s to it to keep the same behaviour as `numpy.tile`.
+  // Thus for a tensor of shape (2, 3, 4, 5), a dims of (2, 2) is treated
+  // as (1, 1, 2, 2).
+  const int64_t size_diff = self.dim() - static_cast<int64_t>(reps.size());
+  if (size_diff > 0){
+    std::vector<int64_t> new_reps(size_diff, 1);
+    for(auto i = decltype(reps.size()){0}; i < reps.size(); ++i){
+      new_reps.emplace_back(reps[i]);
+    }
+    return at::native::repeat(self, IntArrayRef(new_reps));
```

Comment:
Use at::repeat instead of at::native::repeat here and below

Reply:
it doesn't compile with `at::repeat`. Otherwise, I can use `self.repeat(reps)`?

Reply:
Sorry to mislead you. Using the method `self.repeat()` sounds great.